graph-kiota/kiota.md

```diff
@@ -0,0 +1,180 @@
+            enableBackingStore(backingStoreFactory?: BackingStoreFactory | undefined): void;
+        }
+        ````
+        - Serialization interfaces
+        - Authentication: `IAuthProvider`, abstract classes- `anonymousAuthenticationProvider` and `BearerTokenAuthneticationProvider`
+        - Utils
+        - ResponseHandler interface
+        - MiddlewareOption class
+        - RequestInformation
+        - HttpMethod Enum
+
+- Kiota core
+    - This is generic core client library (functionality or purpose similar to Microsoft Graph core libraries)
+    [Kiota core dotnet](https://github.com/microsoft/kiota/tree/main/http/dotnet/httpclient)
+    [Kiota core java](https://github.com/microsoft/kiota/tree/main/http/java/okhttp)
+    [Kiota core typescript](https://github.com/microsoft/kiota/tree/main/http/typescript/fetch)
+    - This library contains the following:
+        -  HttpCore class implementing the abstraction HttpCore interface.
+        -  HttpClientbuilder (.NET, Java) or HTTPClient class (TS) (processes and records the middleware chain used while making requests.)
+        -  Middlewares: AuthHandler, Retry, Redirect, Compression, Caching, Telemetry
+    
+```
+class HttpCore implements kiota.abstractions.HttpCore {
+    constructor(authProvider, parseNodeFactory, serializationWriteFactory, httpClient);
+
+    sendAsync<ModelType extends Parsable>(requestInfo: RequestInformation, type: new() => ModelType, responseHandler: ResponseHandler | undefined): Promise<ModelType>;
+    sendCollectionAsync<ModelType extends Parsable>(requestInfo: RequestInformation, type: new() => ModelType, responseHandler: ResponseHandler | undefined): Promise<ModelType[]>;
+    sendPrimitiveAsync<ResponseType>(requestInfo: RequestInformation, responseType: "string" | "number" | "boolean" | "Date" | "ReadableStream", responseHandler: ResponseHandler | undefined): Promise<ResponseType>;
+    sendNoResponseContentAsync(requestInfo: RequestInformation, responseHandler: ResponseHandler | undefined): Promise<void>;
+    enableBackingStore(backingStoreFactory?: BackingStoreFactory | undefined): void;
+}
+
+class HttpClient{
+    middlewares : Middlewares();
+    send() //calls the MiddlewareChain to make a  request with the native HttpClient;
+}
+```
+
+Kiota Generated API Client (This will be the GraphServiceClient)
+
+```
+class APIClient {
+
+    httpCoreInstance : HttpCore
+
+    SampleRequestBuilder(httpCoreInstance);
+    ExampleRequestBuilder(httpCoreInstance);
+}
+```
+
+- Kiota serializations 
+    - This library contains implementation of the kiota abstractions serialization interfaces
+    - Currently targets `JSON` format.
+    - Other formats to consider:
+        - Multipart payload
+        - Base64 encoded string
+        - text/plain
+        - xml
+
+- Kiota auth providers
+
+### Example of how we use the Kiota libraries:
+
+1. Clone the Kiota repo, build the project and navigate the path containing the `kiota.exe`
```

Comment:
can we refer to this documentation, and improve it if required? https://github.com/microsoft/kiota/#generating-sdks

Reply:
There are language specific pages here https://github.com/microsoft/kiota/tree/main/docs/requiredtools that show how to use the Kiota tooling.  There dotnet one has an end to end experience.  It would be great to do the same for JS/TS